src/emulator/functionsEmulatorShell.ts

```diff
@@ -68,7 +68,13 @@ export class FunctionsEmulatorShell implements FunctionsShellController {
       data,
     };
 
-    this.emu.startFunctionRuntime(trigger.id, trigger.name, getSignatureType(trigger), proto);
+    this.emu.startFunctionRuntime(
+      trigger.id,
+      trigger.name,
+      getSignatureType(trigger),
+      this.emu.getBackends()[0],
```

Comment:
Ahh this feels too magical. One way to avoid the magic would be to extend `FunctionsEmulatorShell` to explictly accept the backend that was used to start the Functions Emulator (which you changed in `src/serve/functions.ts`)

Reply:
Haha, very fair, this was me being lazy because I care very little for functionsEmulatorShell 😬  I'll refactor to be less magical/bad.